HW4.py

```diff
@@ -0,0 +1,120 @@
+"""
+Обязательными будут задача 1 и 2, задачка 3 - когда скучно :)
+
+# ЗАДАЧА-1
+# Написать свой декоратор который будет проверять остаток от деления числа 100 на результат работы функции.
+# Если остаток от деления = 0, вывести сообщение "We are OK!», иначе «Bad news guys, we got {}» остаток от деления.
+
+# ЗАДАЧА-2
+# Написать декоратор который будет выполнять предпроверку типа аргумента который передается в вашу функцию.
+# Если это int, тогда выполнить функцию и вывести результат, если это str(),
+# тогда зарейзить ошибку ValueError (raise ValueError(“string type is not supported”))
+
+Привет, по поводу задачи 2 в декораторах. По условию вы должны там рейзить ошибку, не смущайтесь когда вылетит трейсбек
+с ошибкой  ValueError: string type is not supported. Так и задуманно
+
+Пример рейза ошибок:
+try:
+  f()
+except ValueError:
+  raise ValueError(“string type is not supported”)
+
+# *ЗАДАЧА-3
+# Написать декоратор который будет кешировать значения аргументов и результаты работы вашей функции и записывать
+# его в переменную cache. Если аргумента нет в переменной cache и функция выполняется, вывести сообщение
+# «Function executed with counter = {}, function result = {}» и количество раз сколько эта функция выполнялась.
+# Если значение берется из переменной cache, вывести сообщение «Used cache with counter = {}» и
+# количество раз обращений в cache.
+"""
+
+# ЗАДАЧА-1
+# Написать свой декоратор который будет проверять остаток от деления числа 100 на результат работы функции.
+# Если остаток от деления = 0, вывести сообщение "We are OK!», иначе «Bad news guys, we got {}» остаток от деления.
+
+
+def decorator_division_reminder(division_reminder):
+    def wrapper_around_original_function(*args, **kwargs):
```

Comment:
could be better used @functools.wraps(division_reminder) everywhere in decorators